src/Docker.DotNet/Endpoints/StreamUtil.cs

```diff
@@ -17,15 +17,12 @@ internal static async Task MonitorStreamAsync(Task<Stream> streamTask, DockerCli
             using (var stream = await streamTask)
             {
                 // ReadLineAsync must be cancelled by closing the whole stream.
-                using (cancel.Register(() => stream.Dispose()))
+                using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
                 {
-                    using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
+                    string line;
+                    while ((line = await reader.ReadLineAsync()) != null && !cancel.IsCancellationRequested)

```

Comment:
I'm still confused by this. How does calling cancel unblock a pending `ReadLineAsync` call? This code seems to be required to wait for the next line before it would fall out of the loop.

Reply:
It doesn't stop the current `ReadLineAsync`. It is a `while`-loop; it keeps reading while there is still data and will only stop when it's done reading all the data - which is never because it is a monitoring stream which keeps reporting back data from Docker. The new code `cancel.IsCancellationRequested` is add so that when the process is cancelled: it finishes the current `ReadLineAsync` and quits the `while`-loop.